yt/frontends/amrex/fields.py

```diff
@@ -539,18 +539,22 @@ def __str__(self) -> str:
             f"{element}{count if count > 1 else ''}" for element, count in self._spec
         )
 
+    @staticmethod
+    def _tex_escape(s: str) -> str:
+        return s.replace("_", r"\_")
```

Comment:
I can see this snowballing into a subtle bug where a single underscore would be escaped multiple times instead of once. I'm sure we make this method idempotent using a regexp:

```suggestion
        # replace '_' with '\_' once, but leave alone already-escaped occurrences
        return re.sub(r"[^\\]_",  r"\\_", s)
```